nautobot/extras/models/models.py

```diff
@@ -190,47 +190,7 @@ def get_config_context(self):
-            # Annotation not available, so fall back to manually querying for the config context
-            config_context_data = ConfigContext.objects.get_for_object(self).values_list("data", flat=True)
-        else:
-            config_context_data = self.config_context_data or []
-            # Device and VirtualMachine's Location has its own ConfigContext and its parent Locations' ConfigContext, if any, should
-            # also be applied. However, since moving from mptt to django-tree-queries https://github.com/nautobot/nautobot/issues/510,
-            # we lost the ability to query the ancestors for a particular tree node for subquery https://github.com/matthiask/django-tree-queries/issues/54.
-            # So instead of constructing the location related query in ConfigContextModelQueryset._get_config_context_filters(), which is complicated across databases
-            # We append the missing parent location query here as a patch.
-            location_config_context_queryset = ConfigContext.objects.none()
-            if self._meta.model_name == "device":
-                location_config_context_queryset = ConfigContext.objects.filter(
-                    locations__in=self.location.ancestors(include_self=True)
-                ).distinct()
-            else:
-                if self.cluster and self.cluster.location:
-                    location_config_context_queryset = ConfigContext.objects.filter(
-                        locations__in=self.cluster.location.ancestors(include_self=True)
-                    ).distinct()
-
-            tenant_group_config_context_queryset = ConfigContext.objects.none()
-            if self._meta.model_name == "device" and self.tenant and self.tenant.tenant_group:
-                tenant_group_config_context_queryset = ConfigContext.objects.filter(
-                    tenant_groups__in=self.tenant.tenant_group.ancestors(include_self=True)
-                ).distinct()
-            else:
-                if self.cluster and self.cluster.tenant and self.cluster.tenant.tenant_group:
-                    tenant_group_config_context_queryset = ConfigContext.objects.filter(
-                        tenant_groups__in=self.cluster.tenant.tenant_group.ancestors(include_self=True)
-                    ).distinct()
-
-            # Annotation has keys "weight" and "name" (used for ordering) and "data" (the actual config context data)
-            for cc in location_config_context_queryset:
-                config_context_data.append({"data": cc.data, "name": cc.name, "weight": cc.weight})
-            for cc in tenant_group_config_context_queryset:
-                config_context_data.append({"data": cc.data, "name": cc.name, "weight": cc.weight})
-            config_context_data = [
-                c["data"] for c in sorted(config_context_data, key=lambda k: (k["weight"], k["name"]))
-            ]
+        config_context_data = ConfigContext.objects.get_for_object(self).values_list("data", flat=True)
```

Comment:
The easiest fix is to use `get_for_object` for all cases because the logic is correct. Might be a performance penalty?

Reply:
Definitely a performance hit since we're no longer using the precalculated `self.config_context_data`. Not sure if that's an acceptable loss now that config contexts are no longer included in the REST API by default?